core/src/impl/Kokkos_HostThreadTeam.hpp

```diff
@@ -864,55 +864,79 @@ KOKKOS_INLINE_FUNCTION
 
 //----------------------------------------------------------------------------
 
-template <typename iType, class Closure, class Member>
+template <typename iType, class Closure, class Member,
+          class ValueType = typename Kokkos::Impl::FunctorAnalysis<
+              Kokkos::Impl::FunctorPatternInterface::SCAN, void, Closure,
+              void>::value_type>
```

Comment:
Why does `ValueType` need a default type? It should be deduced from the third function argument, right?
It makes more sense to have a `static_assert` checking the compatibility of the return value's type and the functor's value type.
```suggestion
          class ValueType>
```

Reply:
That's good point. I updated that to use `static_assert` to check compatibility of the types. In current changes I needed to add a default value for `FunctorAnalysis` because it couldn't deduce the type when there was a generic functor used. (`GenericExclusiveScanFunctor` from the tests)